src/org/traccar/protocol/SviasProtocolDecoder.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.helper.PatternBuilder;
+
+import java.net.SocketAddress;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+import org.traccar.DeviceSession;
+import org.traccar.helper.Parser;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.Position;
+
+public class SviasProtocolDecoder extends BaseProtocolDecoder {
+
+    private static final String MSG_KEEPALIVE = "@";
+
+    public SviasProtocolDecoder(SviasProtocol protocol) {
+        super(protocol);
+    }
+
+    private static final Pattern PATTERN = new PatternBuilder()
+            .text("[")       // delimiter init
+            .number("(dddd),") // version hardware
+            .number("(dddd),") // version software
+            .number("(d+),") // counter
+            .number("(d+),") // imei
+            .any()            // model or hourmeter
+            .number("(d+),") // date (yyyymmdd)
+            .number("(d+),") // time (hhmmss)
+            .number("(-?d+),") // longitude
+            .number("(-?d+),") // latitude
+            .number("(d+),") // speed
+            .number("(d+),") // course
+            .number("(d+),") // odometer
+            .number("(d+),") // input
+            .number("(d+),") // output / status
+            .number("(d+),") // flag pack input 1
+            .number("(d+),") // flag pack input 2
+            .number("(d+),") // main power voltage
+            .number("(d+),") // percentual power internal battery
+            .number("(d+),") // RSSID
+            .any()
+            .compile();
+
+    private double convertCoordinates(long v) {
+        return Double.valueOf(((float) ((((float) v / 1.0E7F)

```

Comment:
I believe this can be simplified.

Reply:
I guarantee you that it can be.

Reply:
Give it a hand! I've exhausted my neurons! haha
Only this one to complete!

Reply:
Can I see documentation for it?

Reply:
![image](https://user-images.githubusercontent.com/600484/41007589-df58a46a-68fc-11e8-9ffa-6fb3dfcf65f1.png)
In portuguese!! 

In the device manual there is no talking about this calculation. The analyst of the company that sent an email with the formula.

Reply:
Do you want to see the entire document?

It's all in Brazilian Portuguese.

Reply:
Where exactly did you get your complicated formula? Just use `HEM_DEG_MIN` format.